networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/rpc/core/encodings/LengthPrefixedEncodingTest.java

```diff
@@ -41,65 +42,109 @@
 
   @Test
   public void decodePayload_shouldReturnErrorWhenLengthPrefixIsTooLong() {
-    assertThatThrownBy(
-            () ->
-                encoding.decodePayload(
-                    inputStream("0xAAAAAAAAAAAAAAAAAAAA80"), StatusMessage.class))
+    ByteBuf input = inputByteBuffer("0xAAAAAAAAAAAAAAAAAAAA80");
+    assertThatThrownBy(() -> encoding.createDecoder(StatusMessage.class).decodeOneMessage(input))
```

Comment:
(nit) Could probably improve readability by just having an instance variable for the status decoder that we use throughout:
```suggestion
    assertThatThrownBy(() -> statusDecoder.decodeOneMessage(input))
```